packages/opentelemetry-propagator-jaeger/src/JaegerHttpTracePropagator.ts

```diff
@@ -55,31 +59,67 @@ export class JaegerHttpTracePropagator implements TextMapPropagator {
 
   inject(context: Context, carrier: unknown, setter: TextMapSetter) {
     const spanContext = getSpanContext(context);
-    if (!spanContext || isInstrumentationSuppressed(context)) return;
-
-    const traceFlags = `0${(spanContext.traceFlags || TraceFlags.NONE).toString(
-      16
-    )}`;
-
-    setter.set(
-      carrier,
-      this._jaegerTraceHeader,
-      `${spanContext.traceId}:${spanContext.spanId}:0:${traceFlags}`
-    );
+    const baggage = getBaggage(context);
+    if (spanContext && isInstrumentationSuppressed(context) === false) {
+      const traceFlags = `0${(
+        spanContext.traceFlags || TraceFlags.NONE
+      ).toString(16)}`;
+
+      setter.set(
+        carrier,
+        this._jaegerTraceHeader,
+        `${spanContext.traceId}:${spanContext.spanId}:0:${traceFlags}`
+      );
+    }
+
+    if (baggage) {
+      for (const [key, entry] of baggage.getAllEntries()) {
+        setter.set(
+          carrier,
+          `${UBER_BAGGAGE_HEADER_PREFIX}-${key}`,
+          encodeURIComponent(entry.value)
+        );
+      }
+    }
   }
 
   extract(context: Context, carrier: unknown, getter: TextMapGetter): Context {
     const uberTraceIdHeader = getter.get(carrier, this._jaegerTraceHeader);
     const uberTraceId = Array.isArray(uberTraceIdHeader)
       ? uberTraceIdHeader[0]
       : uberTraceIdHeader;
-
-    if (typeof uberTraceId !== 'string') return context;
-
-    const spanContext = deserializeSpanContext(uberTraceId);
-    if (!spanContext) return context;
-
-    return setSpanContext(context, spanContext);
+    const baggageValues = getter
+      .keys(carrier)
+      .filter(key => key.includes(UBER_BAGGAGE_HEADER_PREFIX))
+      .map(key => {
+        const value = getter.get(carrier, key);
+        return {
+          key: key.replace(`${UBER_BAGGAGE_HEADER_PREFIX}-`, ''),
```

Comment:
Personally I find `substring()` more intuitive here - in special if we change to `startWith()` above.

Reply:
substring is likely more optimized implementation as well due to not needing a new string allocation